formal-models/README.md

```diff
@@ -0,0 +1,135 @@
+## dBFT formal models
+
+This section contains the set of dBFT's formal specifications written in
+[TLA⁺](https://lamport.azurewebsites.net/tla/tla.html) language. The models
+describe the core algorithm logic represented in a high-level way and may be used
+to illustrate some basic dBFT concepts and to validate the algorithm in terms of
+liveness and fairness. It must be noted that presented models do not precisely
+follow the dBFT implementation presented in the repository and may omit some
+implementation details in favor of the specification simplicity and the
+fundamental philosophy of the TLA⁺. However, the presented models directly
+reflect some liveness problems dBFT 2.0 has; the models can and aimed to be used
+for the dBFT 2.0 liveness evaluation and further algorithm improvements.
+
+We will highly appreciate any contributions, questions and discussions on the
+presented models.
+
+## Models
+
+### Basic dBFT 2.0 model
+
+This specification is a basis that was taken for the further algorithm
+investigation. We recommend to begin acquaintance with the dBFT models from this
+one.
+ 
+The specification describes the process of a single block acceptance: the set of
+resource managers `RM` (which is effectively a set of consensus nodes)
+communicating via the shared consensus message pool `msgs` and taking the
+decision in a few consensus rounds (views). Each consensus node has its own state
+at each step of the behaviour. Consensus node may send a consensus message by
+adding it to the `msgs` pool. To perform the transition between states the
+consensus node must send a consensus message or there must be a particular set of
+consensus messages in the shared message pool required for a particular
+transition.
+
+Here's the scheme of transitions between consensus node states:
+
+![Basic dBFT model transitions scheme](./.github/dbft.png)
```

Comment:
Do we have some editable source for it?

Reply:
Yes, it's in the https://app.creately.com/d/cAEwPtsWZGV/edit, but I've just realize that its free plan is too restricted, so we probably need to choose some other plotting tool. Anyway, it was just a quick sketch, so it doesn't take a lot of time to draw it once again. I actually have the whole set of flow diagrams drawn in the notebook.

Reply:
Here's the file: https://drive.google.com/file/d/1ugE3jXq5EkmFdI2r5bWRazzM5oZNoLze/view?usp=sharing
and here's the tool: https://app.diagrams.net